drivers/adc/ad463x/ad463x.c

```diff
@@ -868,11 +946,11 @@ int32_t ad463x_remove(struct ad463x_dev *dev)
 
 	if (!dev)
 		return -1;
-#ifndef USE_STANDARD_SPI
+
 	ret = no_os_pwm_remove(dev->trigger_pwm_desc);
 	if (ret != 0)
 		return ret;
```

Comment:
Now we need to check a PWM was initialized before removing it, no?
```c
if (dev->spi_dma_enable || dev->offload_enable) {
        ret = no_os_pwm_remove(dev->trigger_pwm_desc);
        ...
```

Reply:
actually i had done some thing like this on a different PR in the past,
but after discussions with the reviewer, it seems that in noOs the duality is
not required as its expected that remove can be called without init being called first
there should be checks for null in every remove.

conclusion was that i should not check if init was called in the remove/error paths.